cmd/internal/find/primary/meta/timePredicate.go

```diff
@@ -39,28 +40,45 @@ func parseTimePredicate(tokens []string) (Predicate, []string, error) {
 		// 'StartTime - timeV'.
 		subFromStartTime = false
 	}
-	p := func(v interface{}) bool {
-		timeV, err := munge.ToTime(v)
-		if err != nil {
-			return false
-		}
-		var diff int64
-		if subFromStartTime {
-			diff = int64(params.StartTime.Sub(timeV))
-		} else {
-			diff = int64(timeV.Sub(params.StartTime))
-		}
-		if diff < 0 {
-			// Time predicates query either the past or the future, but not both.
-			// For example, +1h means "more than one hour ago", which queries the
-			// past. diff < 0 there means timeV is from the future, so the query
-			// doesn't make sense. Similarly, +{1h} means "more than one hour from now",
-			// which queries the future. diff < 0 there means timeV is from the past,
-			// so the query doesn't make sense. Thus, diff < 0 is a time-mismatch.
-			// Time-mismatches always return false.
-			return false
-		}
-		return numericP(diff)
+	return timeP(subFromStartTime, p), tokens[1:], nil
+}
+
+func timeP(subFromStartTime bool, p numeric.Predicate) predicate.Predicate {
+	return &timePredicate{
+		genericPredicate: func(v interface{}) bool {
+			timeV, err := munge.ToTime(v)
+			if err != nil {
+				return false
+			}
+			var diff int64
+			if subFromStartTime {
+				diff = int64(params.StartTime.Sub(timeV))
+			} else {
+				diff = int64(timeV.Sub(params.StartTime))
+			}
+			if diff < 0 {
+				// Time predicates query either the past or the future, but not both.
+				// For example, +1h means "more than one hour ago", which queries the
+				// past. diff < 0 there means timeV is from the future, so the query
+				// doesn't make sense. Similarly, +{1h} means "more than one hour from now",
+				// which queries the future. diff < 0 there means timeV is from the past,
+				// so the query doesn't make sense. Thus, diff < 0 is a time-mismatch.
+				// Time-mismatches always return false.
+				return false
+			}
+			return p(diff)
+		},
+		subFromStartTime: true,
```

Comment:
It looks like `subFromStartTime` is always true, so there's no reason to have it be a parameter of `timePredicate`.

Reply:
Oh, that was a mistake. Thanks for catching it. I'm surprised the tests didn't fail. Will fix.

Reply:
Nevermind, looks like it isn't needed. The `if diff < 0` check already handles time-mismatches. Will remove it.